mopidy_party/static/controller.js

```diff
@@ -153,49 +162,66 @@ angular.module('partyApp', [])
       }
       _index++;
     }
-
     $scope.$apply();
   };
 
   $scope.addTrackResult = function(track){
-
     $scope.tracks.push(track);
-    mopidy.tracklist.filter({'uri': [track.uri]}).done(
+	mopidy.tracklist.filter([{'uri': [track.uri]}]).done(
       function(matches){
         if (matches.length) {
           for (var i = 0; i < $scope.tracks.length; i++)
           {
             if ($scope.tracks[i].uri == matches[0].track.uri)
               $scope.tracks[i].disabled = true;
           }
-          $scope.$apply();
         }
       });
+    $scope.$apply();
```

Comment:
Why is it needed to move the $apply?

Reply:
Good catch. I don't think it makes a difference. It was a part of my debugging for the errors in the debug console of the browser, mentioned in the PR message:

>With browser console I discovered some AngularJS errors regarding nested use of $scope.$apply. Got around them by using the AngularJS $timeout functionality.

I think it is safe to move it back.